pkg/util/subnet/subnet.go

```diff
@@ -142,12 +142,17 @@ func (m *manager) GetAll(ctx context.Context, subnetIds []string) ([]*mgmtnetwor
 	return subnets, nil
 }
 
-func (m *manager) CreateOrUpdateFromIds(ctx context.Context, subnetIds []string) error {
+func (m *manager) CreateOrUpdateFromIds(ctx context.Context, subnetIds []string, gatewayEnabled bool) error {
 	subnets, err := m.GetAll(ctx, subnetIds)
 	if err != nil {
 		return err
 	}
 
+	// If egress lockdown is enabled, do not add service endpoints to the subnets.
+	if gatewayEnabled {
+		return m.createOrUpdateSubnets(ctx, subnets)
+	}
```

Comment:
It's just a nit, but maybe we can reverse this condition. It reads more like "we are defaulting to no service endpoints" that way.

	if !gatewayEnabled { // if the cluster is old :)
		// apply service endpoints if not already there
	}